### ISSUE TITLE
Support for access to incomplete .tar.gz files

### ISSUE THREAD

**Issue by mxmlnkn** [mxmlnkn/ratarmount#34]
Try this:

```bash
timeout 30 kaggle competitions download -c imagenet-object-localization-challenge
7z x imagenet-object-localization-challenge.zip
gdb --args python3 $( which ratarmount ) imagenet_object_localization_patched2019.tar.gz
    Currently at position 31223576 of 190691051 (16.37%). Estimated time remaining with current rate: 0 min 46 s, with average rate: 2 min 45 s.
    Currently at position 148666663 of 190691051 (77.96%). Estimated time remaining with current rate: 0 min 0 s, with average rate: 0 min 9 s.
```

Presumably any other truncated .tar.gz will do, though. Backtrace:

```c++
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff6a285c8 in zran_read () from /home/user/.local/lib/python3.8/site-packages/indexed_gzip/indexed_gzip.cpython-38-x86_64-linux-gnu.so
(gdb) bt
#0  0x00007ffff6a285c8 in zran_read () from /home/user/.local/lib/python3.8/site-packages/indexed_gzip/indexed_gzip.cpython-38-x86_64-linux-gnu.so
#1  0x00007ffff6a20923 in ?? () from /home/user/.local/lib/python3.8/site-packages/indexed_gzip/indexed_gzip.cpython-38-x86_64-linux-gnu.so
#2  0x00007ffff6a093df in ?? () from /home/user/.local/lib/python3.8/site-packages/indexed_gzip/indexed_gzip.cpython-38-x86_64-linux-gnu.so
#3  0x00000000005f2246 in _PyObject_MakeTpCall ()
#4  0x00000000005084f3 in ?? ()
#5  0x00000000005ef1f1 in ?? ()
#6  0x00000000005efd2c in PyObject_CallMethodObjArgs ()
#7  0x0000000000648ced in ?? ()
#8  0x0000000000648dc2 in ?? ()
#9  0x0000000000649dbd in ?? ()
#10 0x0000000000502323 in ?? ()
#11 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#12 0x00000000005f19cb in _PyFunction_Vectorcall ()
#13 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#14 0x00000000005f19cb in _PyFunction_Vectorcall ()
#15 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#16 0x00000000005f19cb in _PyFunction_Vectorcall ()
#17 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#18 0x00000000005654d2 in _PyEval_EvalCodeWithName ()
#19 0x00000000005f1bc5 in _PyFunction_Vectorcall ()
#20 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#21 0x00000000005f19cb in _PyFunction_Vectorcall ()
#22 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#23 0x00000000004fe210 in ?? ()
#24 0x00000000005675bd in _PyEval_EvalFrameDefault ()
#25 0x00000000005654d2 in _PyEval_EvalCodeWithName ()
#26 0x00000000005f1bc5 in _PyFunction_Vectorcall ()
#27 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#28 0x00000000005654d2 in _PyEval_EvalCodeWithName ()
#29 0x00000000005f1bc5 in _PyFunction_Vectorcall ()
#30 0x0000000000597ff8 in ?? ()
#31 0x00000000005f21a7 in _PyObject_MakeTpCall ()
#32 0x000000000056cbdd in _PyEval_EvalFrameDefault ()
#33 0x00000000005f19cb in _PyFunction_Vectorcall ()
--Type <RET> for more, q to quit, c to continue without paging--
#34 0x0000000000567325 in _PyEval_EvalFrameDefault ()
#35 0x00000000005654d2 in _PyEval_EvalCodeWithName ()
#36 0x00000000005f1bc5 in _PyFunction_Vectorcall ()
#37 0x00000000005671fd in _PyEval_EvalFrameDefault ()
#38 0x00000000005654d2 in _PyEval_EvalCodeWithName ()
#39 0x00000000005f1bc5 in _PyFunction_Vectorcall ()
#40 0x0000000000597ff8 in ?? ()
#41 0x00000000005f21a7 in _PyObject_MakeTpCall ()
#42 0x000000000056cbdd in _PyEval_EvalFrameDefault ()
#43 0x00000000006ab288 in ?? ()
#44 0x00000000005671fd in _PyEval_EvalFrameDefault ()
#45 0x00000000005654d2 in _PyEval_EvalCodeWithName ()
#46 0x0000000000686d53 in PyEval_EvalCode ()
#47 0x0000000000676101 in ?? ()
#48 0x000000000067617f in ?? ()
#49 0x0000000000676237 in PyRun_FileExFlags ()
#50 0x00000000006782ba in PyRun_SimpleFileExFlags ()
#51 0x00000000006af5ce in Py_RunMain ()
#52 0x00000000006af959 in Py_BytesMain ()
#53 0x00007ffff7dd90b3 in __libc_start_main (main=0x4ec640 <main>, argc=3, argv=0x7fffffffdca8, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7fffffffdc98) at ../csu/libc-start.c:308
#54 0x00000000005f69be in _start ()
```

Because it is a segfault, it doesn't seem like I can catch any exception. Also the error comes from `indexed_gzip`, so I guess I'd have to ask there.

Edit:

This seems to be another issue. I can't reproduce it with e.g. the first 1MB of firefox-2.tar.gz.

Event the system gzip prints a warning:

gzip -d -k imagenet_object_localization_patched2019.tar.gz 
gzip: imagenet_object_localization_patched2019.tar.gz: invalid compressed data--format violated

It might be that kaggle is downloading in parallel and therefore the end of the gz might have some patches containing only 0 bytes or something like that.
